galaxy_importer/loaders.py

```diff
@@ -134,29 +170,30 @@ def handle_nested_tables(obj, table_key):
                 for row in obj[table_key]:
                     handle_nested_tables(row, table_key)
 
-        doc = data[self.name].get('doc')
+        doc = data.get('doc', None)
         if doc and 'options' in doc.keys() and isinstance(doc['options'], dict):
             doc['options'] = dict_to_named_list(doc['options'])
             for d in doc['options']:
                 handle_nested_tables(d, table_key='suboptions')
 
-        ret = data[self.name].get('return')
+        ret = data.get('return', None)
         if ret and isinstance(ret, dict):
-            data[self.name]['return'] = dict_to_named_list(ret)
-            for d in data[self.name]['return']:
+            data['return'] = dict_to_named_list(ret)
+            for d in data['return']:
                 handle_nested_tables(d, table_key='contains')
 
         return data
 
     def _run_ansible_doc(self):
         cmd = [
+            'env', f'ANSIBLE_COLLECTIONS_PATHS={self.tmp_dir}',
             'ansible-doc',
+            self.fq_name,
             '--type', self.content_type.value,
-            '-M', os.path.dirname(self.rel_path),
-            self.name,
-            '--json']
+            '--json',
+        ]
         self.log.debug('CMD: {}'.format(' '.join(cmd)))
-        proc = Popen(cmd, cwd=self.root, stdout=PIPE, stderr=PIPE)
+        proc = Popen(' '.join(cmd), cwd=self.root, stdout=PIPE, stderr=PIPE, shell=True)
```

Comment:
Why the change to join the cmd arg list to a string here? Keeping popen cmd args in list form tends to help avoid a lot of common problems like quoting/escaping. Also helps avoid potential security issues caused by things like user provided strings doing things like fq_name = 'foo; rm -rf /important_stuff;' and creating a cmd like `env /tmp/whatever ansible-doc foo; rm -rf /important_stuff; --type module some_module_name --json`.

Especially when combined with 'shell=True'.  Why is the `shell=True` needed?

Reply:
in getting `ansible-doc` to recognize the path environment variable and find `doc_fragments` modules, these were the changes I added. I will double check if either or both of those edits (`join` and `shell=True`) are needed.

Reply:
things are working without those edits, so I have removed them